evm/src/modules/IntentGatewayV2.sol

```diff
@@ -0,0 +1,842 @@
+                }
+
+                _orders[commitment][token] += requiredAmount;
+            }
+
+            // Execute transfer calls from call dispatcher
+            ICallDispatcher(callDispatcher).dispatch(abi.encode(transferCalls));
+        } else {
+            uint256 inputsLen = order.inputs.length;
+
+            for (uint256 i = 0; i < inputsLen; i++) {
+                if (order.inputs[i].amount == 0) revert InvalidInput();
+                address token = address(uint160(uint256(order.inputs[i].token)));
+                if (token == address(0)) {
+                    // native token
+                    if (msgValue < order.inputs[i].amount) revert InsufficientNativeToken();
+                    msgValue -= order.inputs[i].amount;
+                } else {
+                    IERC20(token).safeTransferFrom(msg.sender, address(this), order.inputs[i].amount);
+                }
+
+                _orders[commitment][token] += order.inputs[i].amount;
+            }
+        }
+
+        if (order.fees > 0) {
+            // escrow fees
+            address feeToken = IDispatcher(hostAddr).feeToken();
+            if (msgValue > 0) {
+                address uniswapV2 = IDispatcher(hostAddr).uniswapV2Router();
+                address WETH = IUniswapV2Router02(uniswapV2).WETH();
+                address[] memory path = new address[](2);
+                path[0] = WETH;
+                path[1] = IDispatcher(hostAddr).feeToken();
+                IUniswapV2Router02(uniswapV2).swapExactETHForTokens{value: msgValue}(
+                    order.fees,
+                    path,
+                    address(this),
+                    block.timestamp
+                );
+            } else {
+                IERC20(feeToken).safeTransferFrom(msg.sender, address(this), order.fees);
+            }
+
+            _orders[commitment][TRANSACTION_FEES] = order.fees;
+        }
+
+        _nonce += 1;
+        emit OrderPlaced({
+            user: order.user,
+            sourceChain: order.sourceChain,
+            destChain: order.destChain,
+            deadline: order.deadline,
+            nonce: order.nonce,
+            fees: order.fees,
+            outputs: order.outputs,
+            inputs: order.inputs,
+            callData: order.callData
+        });
+    }
+
+    /**
+     * @notice Fills an order with the specified options.
+     * @param order The order to be filled.
+     * @param options The options to be used when filling the order.
+     * @dev This function is payable and can accept Ether.
+     */
+    function fillOrder(Order calldata order, FillOptions memory options) public payable {
+        address hostAddr = host();
+        // Ensure the order is being filled on the correct chain
+        if (keccak256(order.destChain) != keccak256(IDispatcher(hostAddr).host())) revert WrongChain();
+
+        // Ensure the order has not expired
+        if (order.deadline < block.number) revert Expired();
```

Comment:
Missing check for zero deadline

Reply:
Actually for limit orders, we should use `type(uint256).max` so they never expire